components/iotserver-mgt/org.wso2.carbon.device.mgt.iot.common/src/main/java/org/wso2/carbon/device/mgt/iot/common/transport/mqtt/MqttPublisher.java

```diff
@@ -0,0 +1,130 @@
+ * the settings from the default xml org.wso2.carbon.device.mgt.iot.common.devicecloud.org.wso2.carbon.device.mgt.iot.common.config.server.configs
+ * file -
+ * /resources/conf/device-controls/devicecloud-config.xml
+ */
+public class MqttPublisher implements MqttCallback {
+
+	private static final Log log = LogFactory.getLog(MqttPublisher.class);
+
+	private String mqttEndpoint;
+	private String mqttUsername;
+	private String mqttPassword;
+	private boolean mqttEnabled = false;
+
+	public MqttPublisher() {
+	}
+
+	public void initControlQueue() throws DeviceControllerException {
+		mqttEndpoint = MqttConfig.getInstance().getMqttQueueEndpoint();
+		mqttUsername = MqttConfig.getInstance().getMqttQueueUsername();
+		mqttPassword = MqttConfig.getInstance().getMqttQueuePassword();
+		mqttEnabled = MqttConfig.getInstance().isEnabled();
+	}
+
+	public void publish(String publishClientId, String publishTopic, byte[] payload)
+			throws DeviceControllerException {
+
+		if (mqttEnabled) {
+			MqttClient client;
+			MqttConnectOptions options;
+
+			if (publishClientId.length() > 24) {
+				String errorString =
+						"No of characters '" + publishClientId.length() + "' for ClientID: '" + publishClientId +
+								"' is invalid (should be less than 24, hence please provide a " +
+								"simple " +
+
+
+								"'owner' tag)";
+				log.error(errorString);
+				throw new DeviceControllerException(errorString);
+			} else {
+				log.info("No of Characters " + publishClientId.length() + " for ClientID : '" + publishClientId +
+								 "' is acceptable");
+			}
+
+			try {
+				client = new MqttClient(mqttEndpoint,publishClientId);
+				options = new MqttConnectOptions();
+				options.setWill("iotDevice/clienterrors", "crashed".getBytes(UTF_8), 2, true);
+				client.setCallback(this);
+				client.connect(options);
+
+				client.publish(publishTopic, payload, 0, true);
+
+				if (log.isDebugEnabled()) {
+					log.debug("MQTT Client successfully published to topic: " + publishTopic +
+									  ", with payload - " + payload);
+				}
+				client.disconnect();
+			} catch (MqttException ex) {
+				String errorMsg =
+						"MQTT Client Error" + "\n\tReason:  " + ex.getReasonCode() +
+								"\n\tMessage: " +
+								ex.getMessage() + "\n\tLocalMsg: " + ex.getLocalizedMessage() +
+								"\n\tCause: " + ex.getCause() + "\n\tException: " + ex;
+
+				log.error(errorMsg, ex);
+				throw new DeviceControllerException(errorMsg, ex);
+			}
+		} else {
+			log.warn("MQTT <Enabled> set to false in 'devicecloud-config.xml'");
+		}
+	}
+
+	@Override
+	public void connectionLost(Throwable arg0) {
+		log.error("Connection to MQTT Endpoint Lost");
+	}
+
+
+	@Override
+	public void deliveryComplete(IMqttDeliveryToken topic) {
+		log.info("Published topic: '" + topic.getTopics()[0] + "' successfully to client: '" +
+						 topic.getClient().getClientId() + "'");
+	}
+
+	@Override
+	public void messageArrived(String arg0, MqttMessage arg1) throws Exception {
```

Comment:
this class implements MqttCallback hence overriding the method. I agree Log should be in debug/trace level. Since this class does not exist in current repo, cannot fix.